### PR TITLE
chore: the 500 response code in the upstream.lua file was changed to 503 response code

### LINKED ISSUE #3374: Modify the 500 response code in the upstream.lua file to 503 response code
### Issue description

The `500` response code should be changed to a `503` response code. Because this is a situation where upstream cannot work.
Code link: [https://github.com/apache/apisix/blob/master/apisix/upstream.lua#L125-L168 ](https://github.com/apache/apisix/blob/master/apisix/upstream.lua#L125-L168)

### Environment

* apisix version (cmd: `apisix version`):
* OS: (cmd: `uname -a`)
* OpenResty / Nginx version: (cmd: `nginx -V` or `openresty -V`)

### PR DESCRIPTION
**Pull request by Firstsawyou**
### What this PR does / why we need it:

fix #3374

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Review by membphis — APPROVED** [review]
LGTM

**Comment by tokers** [comment]
Does the broken compatibility really  no matter?

**Review by moonming — CHANGES_REQUESTED** [review]
why change this？

File: apisix/upstream.lua [hunk 1]
```diff
@@ -131,21 +131,21 @@ function _M.set_by_route(route, api_ctx)
 
     local up_conf = api_ctx.matched_upstream
     if not up_conf then
-        return 500, "missing upstream configuration in Route or Service"
+        return 503, "missing upstream configuration in Route or Service"
```
Comment (moonming) [inline apisix/upstream.lua:134]: no test case cover

**Review by moonming — CHANGES_REQUESTED** [review]

**Comment by Firstsawyou** [comment]
> why change this？

I think the use of the 500 response code in the code is unreasonable, because this is not caused by a code program error, but because there is no upstream available. So it is more reasonable to use the 503 response code.

**Comment by Firstsawyou** [comment]
> Does the broken compatibility really no matter?

This will only affect some test cases.

**Comment by moonming** [comment]
> > why change this？
> 
> 
> 
> I think the use of the 500 response code in the code is unreasonable, because this is not caused by a code program error, but because there is no upstream available. So it is more reasonable to use the 503 response code.

why will be no upstream available？

**Comment by moonming** [comment]
> > Does the broken compatibility really no matter?
> 
> 
> 
> This will only affect some test cases.

of course not.  This is too dangerous, you don’t know the scope of the impact

**Comment by tokers** [comment]
You don't know whether people rely on the status code. @Firstsawyou

**Comment by Firstsawyou** [comment]
> You don't know whether people rely on the status code. @Firstsawyou

got it, this is something to consider.
But these 500 response codes are inappropriate for use here, we need to fix them.
This problem needs to be solved sooner or later, and it would be better for us to solve these problems earlier.

File: apisix/upstream.lua (hunk 1 above)
Comment (Firstsawyou) [reply apisix/upstream.lua:134]: updated

**Comment by tokers** [comment]
> > You don't know whether people rely on the status code. @Firstsawyou
> 
> got it, this is something to consider.
> But these 500 response codes are inappropriate for use here, we need to fix them.
> This problem needs to be solved sooner or later, and it would be better for us to solve these problems earlier.

we call them historical problems, we have to be compatible with them. Even if you'd like to change it, sending a statement and claiming that we'll change these status codes from `500` to `503` once a.b version is released, which gives some time to user to do the change.

**Comment by moonming** [comment]
@Firstsawyou  Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

**Comment by Firstsawyou** [comment]
> @Firstsawyou Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

Okay.

**Comment by tokers** [comment]
So i think this PR can be closed now, just reopen it when necessary.

**Pull request closed without merging**